commit ca7142ade54d91e44fdffd5df84bc9fea1043a15
Author: Tor Norbye <tnorbye@google.com>
Date:   Fri Jun 10 08:12:27 2016 -0700

    Remove closed-source modules (C++ etc) from build scripts
    
    This makes the AOSP push push a tag that builds AOSP Studio correctly.
    
    Change-Id: I64e86c572f62f94eb1764d2ed5838039447f3fe6

diff --git a/.idea/modules.xml b/.idea/modules.xml
index f81274e..7c73590 100644
--- a/.idea/modules.xml
+++ b/.idea/modules.xml
@@ -13,13 +13,11 @@
       <module fileurl="file://$PROJECT_DIR$/../adt/idea/adt-branding/adt-branding.iml" filepath="$PROJECT_DIR$/../adt/idea/adt-branding/adt-branding.iml" group="android" />
       <module fileurl="file://$PROJECT_DIR$/platform/analysis-api/analysis-api.iml" filepath="$PROJECT_DIR$/platform/analysis-api/analysis-api.iml" group="platform" />
       <module fileurl="file://$PROJECT_DIR$/platform/analysis-impl/analysis-impl.iml" filepath="$PROJECT_DIR$/platform/analysis-impl/analysis-impl.iml" group="platform" />
-      <module fileurl="file://$PROJECT_DIR$/../vendor/google/analytics/analytics.iml" filepath="$PROJECT_DIR$/../vendor/google/analytics/analytics.iml" group="android" />
       <module fileurl="file://$PROJECT_DIR$/../adt/idea/android/android.iml" filepath="$PROJECT_DIR$/../adt/idea/android/android.iml" group="android" />
       <module fileurl="file://$PROJECT_DIR$/../base/annotations/android-annotations.iml" filepath="$PROJECT_DIR$/../base/annotations/android-annotations.iml" group="android/sdktools" />
       <module fileurl="file://$PROJECT_DIR$/../adt/idea/android/common/android-common.iml" filepath="$PROJECT_DIR$/../adt/idea/android/common/android-common.iml" group="android" />
       <module fileurl="file://$PROJECT_DIR$/../adt/idea/android-gradle-jps/android-gradle-jps.iml" filepath="$PROJECT_DIR$/../adt/idea/android-gradle-jps/android-gradle-jps.iml" group="android" />
       <module fileurl="file://$PROJECT_DIR$/../adt/idea/android/jps-plugin/android-jps-plugin.iml" filepath="$PROJECT_DIR$/../adt/idea/android/jps-plugin/android-jps-plugin.iml" group="android" />
-      <module fileurl="file://$PROJECT_DIR$/../vendor/google/android-ndk/android-ndk.iml" filepath="$PROJECT_DIR$/../vendor/google/android-ndk/android-ndk.iml" group="plugins/Android" />
       <module fileurl="file://$PROJECT_DIR$/../adt/idea/android/rt/android-rt.iml" filepath="$PROJECT_DIR$/../adt/idea/android/rt/android-rt.iml" group="android" />
       <module fileurl="file://$PROJECT_DIR$/platform/annotations/annotations.iml" filepath="$PROJECT_DIR$/platform/annotations/annotations.iml" group="platform" />
       <module fileurl="file://$PROJECT_DIR$/platform/annotations/common/annotations-common.iml" filepath="$PROJECT_DIR$/platform/annotations/common/annotations-common.iml" group="platform" />
@@ -29,6 +27,7 @@
       <module fileurl="file://$PROJECT_DIR$/jps/antLayout/antlayout.iml" filepath="$PROJECT_DIR$/jps/antLayout/antlayout.iml" group="jps" />
       <module fileurl="file://$PROJECT_DIR$/plugins/google-app-engine/runtime/appEngine-runtime.iml" filepath="$PROJECT_DIR$/plugins/google-app-engine/runtime/appEngine-runtime.iml" group="plugins/GAE" />
       <module fileurl="file://$PROJECT_DIR$/../base/asset-studio/assetstudio.iml" filepath="$PROJECT_DIR$/../base/asset-studio/assetstudio.iml" group="android/sdktools" />
+      <module fileurl="file://$PROJECT_DIR$/../base/apkparser/binary-resources/binary-resources.iml" filepath="$PROJECT_DIR$/../base/apkparser/binary-resources/binary-resources.iml" group="android/sdktools" />
       <module fileurl="file://$PROJECT_DIR$/platform/boot/boot.iml" filepath="$PROJECT_DIR$/platform/boot/boot.iml" group="platform" />
       <module fileurl="file://$PROJECT_DIR$/platform/bootstrap/bootstrap.iml" filepath="$PROJECT_DIR$/platform/bootstrap/bootstrap.iml" group="platform" />
       <module fileurl="file://$PROJECT_DIR$/build/build.iml" filepath="$PROJECT_DIR$/build/build.iml" />
@@ -37,14 +36,8 @@
       <module fileurl="file://$PROJECT_DIR$/platform/built-in-server-api/built-in-server-api.iml" filepath="$PROJECT_DIR$/platform/built-in-server-api/built-in-server-api.iml" group="platform" />
       <module fileurl="file://$PROJECT_DIR$/platform/built-in-server/built-in-server-tests.iml" filepath="$PROJECT_DIR$/platform/built-in-server/built-in-server-tests.iml" group="platform" />
       <module fileurl="file://$PROJECT_DIR$/../base/chartlib/chartlib.iml" filepath="$PROJECT_DIR$/../base/chartlib/chartlib.iml" group="android/sdktools" />
-      <module fileurl="file://$PROJECT_DIR$/../vendor/intellij/cidr/cidr-common/cidr-common.iml" filepath="$PROJECT_DIR$/../vendor/intellij/cidr/cidr-common/cidr-common.iml" group="java" />
-      <module fileurl="file://$PROJECT_DIR$/../vendor/intellij/cidr/cidr-debugger/cidr-debugger.iml" filepath="$PROJECT_DIR$/../vendor/intellij/cidr/cidr-debugger/cidr-debugger.iml" group="java" />
-      <module fileurl="file://$PROJECT_DIR$/../vendor/intellij/cidr/cidr-ide/cidr-ide.iml" filepath="$PROJECT_DIR$/../vendor/intellij/cidr/cidr-ide/cidr-ide.iml" group="java" />
-      <module fileurl="file://$PROJECT_DIR$/../vendor/intellij/cidr/cidr-lang/cidr-lang.iml" filepath="$PROJECT_DIR$/../vendor/intellij/cidr/cidr-lang/cidr-lang.iml" group="java" />
-      <module fileurl="file://$PROJECT_DIR$/../vendor/intellij/cidr/cidr-lang-dfa/cidr-lang-dfa.iml" filepath="$PROJECT_DIR$/../vendor/intellij/cidr/cidr-lang-dfa/cidr-lang-dfa.iml" group="java" />
-      <module fileurl="file://$PROJECT_DIR$/../vendor/intellij/cidr/cidr-util/cidr-util.iml" filepath="$PROJECT_DIR$/../vendor/intellij/cidr/cidr-util/cidr-util.iml" group="java" />
-      <module fileurl="file://$PROJECT_DIR$/../vendor/intellij/cidr/clion/clion.iml" filepath="$PROJECT_DIR$/../vendor/intellij/cidr/clion/clion.iml" group="java" />
-      <module fileurl="file://$PROJECT_DIR$/../vendor/intellij/cidr/clion-build/clion-build.iml" filepath="$PROJECT_DIR$/../vendor/intellij/cidr/clion-build/clion-build.iml" group="java" />
+      <module fileurl="file://$PROJECT_DIR$/../base/chunkio/chunkio.iml" filepath="$PROJECT_DIR$/../base/chunkio/chunkio.iml" group="android/sdktools" />
+      <module fileurl="file://$PROJECT_DIR$/../base/chunkio-processor/chunkio-processor.iml" filepath="$PROJECT_DIR$/../base/chunkio-processor/chunkio-processor.iml" group="android/sdktools" />
       <module fileurl="file://$PROJECT_DIR$/colorSchemes/colorSchemes.iml" filepath="$PROJECT_DIR$/colorSchemes/colorSchemes.iml" group="platform" />
       <module fileurl="file://$PROJECT_DIR$/plugins/commander/commander.iml" filepath="$PROJECT_DIR$/plugins/commander/commander.iml" group="plugins" />
       <module fileurl="file://$PROJECT_DIR$/../base/common/common.iml" filepath="$PROJECT_DIR$/../base/common/common.iml" group="android/sdktools" />
@@ -101,6 +94,8 @@
       <module fileurl="file://$PROJECT_DIR$/platform/extensions/extensions.iml" filepath="$PROJECT_DIR$/platform/extensions/extensions.iml" group="platform" />
       <module fileurl="file://$PROJECT_DIR$/platform/external-system-api/external-system-api.iml" filepath="$PROJECT_DIR$/platform/external-system-api/external-system-api.iml" group="platform" />
       <module fileurl="file://$PROJECT_DIR$/platform/external-system-impl/external-system-impl.iml" filepath="$PROJECT_DIR$/platform/external-system-impl/external-system-impl.iml" group="platform" />
+      <module fileurl="file://$PROJECT_DIR$/platform/external-system-rt/external-system-rt.iml" filepath="$PROJECT_DIR$/platform/external-system-rt/external-system-rt.iml" group="platform" />
+      <module fileurl="file://$PROJECT_DIR$/../studio/google/cloud/testing/firebase-testing/firebase-testing.iml" filepath="$PROJECT_DIR$/../studio/google/cloud/testing/firebase-testing/firebase-testing.iml" group="plugins/Google" />
       <module fileurl="file://$PROJECT_DIR$/java/compiler/forms-compiler/forms-compiler.iml" filepath="$PROJECT_DIR$/java/compiler/forms-compiler/forms-compiler.iml" group="java/compiler" />
       <module fileurl="file://$PROJECT_DIR$/platform/forms_rt/forms_rt.iml" filepath="$PROJECT_DIR$/platform/forms_rt/forms_rt.iml" group="platform" />
       <module fileurl="file://$PROJECT_DIR$/plugins/git4idea/git4idea.iml" filepath="$PROJECT_DIR$/plugins/git4idea/git4idea.iml" group="plugins/VCS" />
@@ -300,4 +295,4 @@
       <module fileurl="file://$PROJECT_DIR$/plugins/xpath/xslt-rt/xslt-rt.iml" filepath="$PROJECT_DIR$/plugins/xpath/xslt-rt/xslt-rt.iml" group="plugins" />
     </modules>
   </component>
-</project>
\ No newline at end of file
+</project>
diff --git a/build/scripts/dist.gant b/build/scripts/dist.gant
index 023c2c8..3051016 100644
--- a/build/scripts/dist.gant
+++ b/build/scripts/dist.gant
@@ -82,7 +82,6 @@ target(compile: "Compile project") {
 
   projectBuilder.targetFolder = "$out/classes"
   projectBuilder.cleanOutput()
-  bundledJDKs()
   projectBuilder.buildProduction()
   projectBuilder.makeModuleTests(findModule("jps-builders"))
 
diff --git a/build/scripts/layouts.gant b/build/scripts/layouts.gant
index 891b351..0d5f4e4 100644
--- a/build/scripts/layouts.gant
+++ b/build/scripts/layouts.gant
@@ -102,11 +102,6 @@ def layoutFull(String home, String targetDirectory, String patchedDescriptorDir
   ].flatten()
 
   List<String> implementationModules = [platformImplementationModules,
-    "cidr-common",
-    "cidr-debugger",
-    "cidr-lang",
-    "cidr-lang-dfa",
-    "cidr-util",
     "compiler-impl",
     "debugger-impl",
     "dom-impl",
diff --git a/build/scripts/studio_properties.gant b/build/scripts/studio_properties.gant
index 54f4052..b68e97c 100644
--- a/build/scripts/studio_properties.gant
+++ b/build/scripts/studio_properties.gant
@@ -96,10 +96,6 @@ def getProperties(String home, String buildNumber) {
       }
 
       layout(targetDirectory) {
-        dir("${lldbBinTargetRoot}/shared") {
-          fileset(dir: "${lldbBinSrcRoot}/shared")
-        }
-
         dir("gradle") {
           dir("m2repository") {
             fileset(dir: "$studioRepo")
@@ -107,22 +103,6 @@ def getProperties(String home, String buildNumber) {
         }
 
         dir("plugins") {
-          dir("android-ndk") {
-            dir("lib") {
-              jar("android-ndk.jar") {
-                module("android-ndk")
-              }
-            }
-          }
-
-          dir("analytics") {
-            dir("lib") {
-              jar("analytics.jar") {
-                module("analytics")
-              }
-            }
-          }
-
           dir("google-cloud-tools") {
             dir("lib") {
               jar("google-cloud-tools.jar") {
@@ -272,6 +252,31 @@ def getProperties(String home, String buildNumber) {
     }
 
     def customMacLayout(targetDirectory) {
+      ant.copy(file: "$home/$androidRoot/android/lib/libwebp/mac/libwebp_jni64.dylib", tofile: "$targetDirectory/plugins/android/lib/libwebp_jni64.dylib")
+    }
+
+    // Patches plugin.xml to replace the "SNAPSHOT" in the version with build number and product-build.txt to replace "PRODUCT_BUILD"
+    // with the full build number.
+    def updateAswbPlugin() {
+      def aswb = projectBuilder.moduleOutput(findModule("aswb"))
+
+      def pluginFile = new File(aswb + "/META-INF/plugin.xml")
+      if (pluginFile.isFile()) {
+        def text = pluginFile.text
+        text = text.replaceAll("SNAPSHOT", buildNumber);
+        pluginFile.text = text;
+      }
+
+      def productBuildFile = new File(aswb + "/META-INF/product-build.txt")
+      if (productBuildFile.isFile()) {
+        def text = productBuildFile.text
+        text = text.replaceAll("PRODUCT_BUILD", fullBuildNumber());
+        productBuildFile.text = text;
+      }
+
+    }
+
+    def finishLayout(out) {
     }
   }
 }
diff --git a/build_studio.sh b/build_studio.sh
index ecd5b59..f07897e 100755
--- a/build_studio.sh
+++ b/build_studio.sh
@@ -103,12 +103,6 @@ export JDK_18_x64=$JAVA8_HOME
 
 export PATH=$JDK_18_x64/bin:$PATH
 
-$ANT "-Dout=$OUT" "-Dbuild=$BNUM" "-Denable.ui.tests=$UI_TESTS" "-Dinclude.blaze=$BLAZE"
-
-echo "## Copying android-studio distribution files"
-mkdir -p "$DIST"
-cp -Rfv "$OUT"/artifacts/android-studio* "$DIST"/
-cp -Rfv "$OUT"/updater-full.jar "$DIST"/android-studio-updater.jar
-# write the version number into the windows installer dir
-echo $BNUM > ../adt/idea/native/installer/win/version
-(cd ../adt/idea/native/installer/win && zip -r - ".") > "$DIST"/android-studio-bundle-data.zip
+# Temp: Figure out if we can properly default to Java8
+export JAVA_HOME=$JAVA8_HOME
+$ANT "-Dout=$OUT" "-Dbuild=$BNUM" "-Denable.ui.tests=$UI_TESTS"
diff --git a/community-main.iml b/community-main.iml
index 6cfdebc..bbce667 100644
--- a/community-main.iml
+++ b/community-main.iml
@@ -122,6 +122,15 @@
     <orderEntry type="module" module-name="sdk-updates" />
     <orderEntry type="module" module-name="designer" />
     <orderEntry type="module" module-name="analytics" />
-    <orderEntry type="module" module-name="android-ndk" />
+    <orderEntry type="module" module-name="devkit-tests" scope="TEST" />
+    <orderEntry type="module" module-name="diff-tests" scope="TEST" />
+    <orderEntry type="module" module-name="built-in-server-tests" scope="TEST" />
+    <orderEntry type="module" module-name="updater" scope="TEST" />
+    <orderEntry type="module" module-name="coverage" />
+    <orderEntry type="module" module-name="android-gradle-jps" scope="TEST" />
+    <orderEntry type="module" module-name="android-jps-plugin" scope="TEST" />
+    <orderEntry type="module" module-name="script-debugger-ui" scope="TEST" />
+    <orderEntry type="module" module-name="remote-servers-git-java" />
+    <orderEntry type="module" module-name="terminal" />
   </component>
 </module>
